### ISSUE TITLE
Include comment at top of .proto file

### ISSUE THREAD

**Issue by oije2333** [estan/protoc-gen-doc#9]
Feature requests:

If a proto file contains a "/** ... */" comment at the top of the file, that comment should be included in the generated documentation.

**Comment by estan** [comment]
Hi @oije2333,

Thanks for the request. Including default values has been requested before, so that's definitely something I should do. Could you create a separate GitHub issue for that though? It makes it easier to track requests if it's one request per GitHub issue.

The "top of file" comment request... I'll see what I can do. I'm not sure the descriptors provided by the protobuf library has that information (that it's the first comment). But I'll have a look. A bit busy right now though, so it may take a while.

**Comment by ngg** [comment]
+1

If it is not possible with protobuf descriptors, then there might be an optional parameter with a file that contains Markdown text to be included at the top of the documentation.

**Comment by estan** [comment]
I'll investigate whether it's really impossible. If not, I'll add a feature like you say @ngg.

I've opened up a separate issue #11 for the default values.

**Comment by estan** [comment]
Issue #11 is now fixed. I've added support for default values.

I've also added issue #12 for the header file solution suggested by @ngg , since that would be useful anyway, and is not really the same as this issue, since it would be a header for the documentation as a whole (visible under the title), not a per-file description as requested here.

Regarding per-file comments, I think that even if it turns out there's no way to do this through the protobuf plugin API, I should be able to do it by some custom parsing, so I'm keeping this issue open.
